src/Toolkit.Forms/FloorFilter/FloorFilter.cs

```diff
@@ -0,0 +1,647 @@
+            _controller.PropertyChanged += HandleControllerPropertyChanges;
+        }
+
+        /// <inheritdoc/>
+        protected override void OnApplyTemplate()
+        {
+            if (PART_ZoomButton != null)
+            {
+                PART_ZoomButton.Clicked -= HandleZoomButtonClick;
+                PART_ZoomButton = null;
+            }
+
+            if (PART_BrowseButton != null)
+            {
+                PART_BrowseButton.Clicked -= PART_BrowseButton_Clicked;
+                PART_BrowseButton = null;
+            }
+
+            PART_AllButton = null;
+            PART_LevelListContainer = null;
+
+            base.OnApplyTemplate();
+
+            PART_VisibilityWrapper = GetTemplateChild(nameof(PART_VisibilityWrapper)) as View;
+            PART_VisibilityWrapper?.SetValue(IsVisibleProperty, _controller.ShouldDisplayFloorPicker);
+
+            if (GetTemplateChild(nameof(PART_ZoomButton)) is Button zoomButton)
+            {
+                PART_ZoomButton = zoomButton;
+                PART_ZoomButton.Clicked += HandleZoomButtonClick;
+            }
+
+            PART_LevelListView = GetTemplateChild(nameof(PART_LevelListView)) as ListView;
+            PART_BrowseButton = GetTemplateChild(nameof(PART_BrowseButton)) as Button;
+            PART_AllButton = GetTemplateChild(nameof(PART_AllButton)) as Button;
+            PART_LevelListContainer = GetTemplateChild(nameof(PART_LevelListContainer)) as View;
+            PART_LevelListContainer?.SetValue(IsVisibleProperty, false);
+            if (PART_BrowseButton != null)
+            {
+                PART_BrowseButton.Clicked += PART_BrowseButton_Clicked;
+            }
+
+            if (PART_LevelListView != null)
+            {
+                PART_LevelListView.ItemTemplate = LevelDataTemplate;
+                PART_LevelListView.BindingContext = this;
+                PART_LevelListView.SetBinding(ListView.ItemsSourceProperty, new Binding(nameof(AllLevels), BindingMode.OneWay));
+                PART_LevelListView.SetBinding(ListView.SelectedItemProperty, new Binding(nameof(SelectedLevel), BindingMode.TwoWay));
+            }
+        }
+
+        private void PART_BrowseButton_Clicked(object sender, EventArgs e)
+        {
+            if (AllSites?.Count > 1)
+            {
+                NavigateForward(new FloorFilterBrowseSitesPage(this));
+            }
+            else
+            {
+                NavigateForward(new FloorFilterBrowseFacilitiesPage(this, true, false));
+            }
+        }
+
+        internal async void NavigateForward(ContentPage page)
+        {
+            try
+            {
+                await Navigation.PushModalAsync(page);
+                _navigationStackCounter++;
+            }
+            catch (Exception ex)
+            {
+                System.Diagnostics.Debug.WriteLine($"Toolkit exception while navigating - {ex}");
+            }
+        }
```

Comment:
I'm not sure it can. Xamarin.Forms seems to be pretty limited wrt handling popups/floating windows. Without a popup, I could show a separate view side-by-side, but that may not work well on width-constrained devices, and it limits options for how the user can arrange the view.

One option might be to expose a property to hide the browse button and disable the browsing experience. That way, if the developer wants to do a custom implementation of browsing, or not deal with navigation, they can still get most of the value from the control.